### PR TITLE
[4.0] Deprecate unused $iconOver

### PR DESCRIPTION
**Pull request by N6REJ**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/30211

### Summary of Changes
Removes unused $iconOver param.
In 1.5 days this was used as an override ( not hover as stated now ) but has never been implemented in 4.+ so might as well just remove it.  If its determined later that its useful we can always create it because as it is now it does NOTHING.

### Testing Instructions
notice that toolbar icons display.
apply pr.
make sure toolbar icons still display.
hover over icons notice the icons remain.

### Actual result BEFORE applying this Pull Request
nothing noted

### Expected result AFTER applying this Pull Request
no visual changes.

### Documentation Changes Required
Deprecate $iconOver in 5.x

### CODE REVIEW

**Comment by SharkyKZ** [comment]
This can't be removed without rewriting all uses of the method (B/C break) or adding B/C layer based on argument count.

**Comment by N6REJ** [comment]
@SharkyKZ well, the thing is its not even used!!!
https://github.com/joomla/joomla-cms/blob/425de684cdaaa8e5f51d66579ce3f1121018f501/libraries/src/Toolbar/ToolbarHelper.php#L102-L111
if you add $iconOver to the appendButton function it becomes a text value in place of the normal text.. i.e. everything is out of sync.
in 3.x it an override NOT hover ( which makes sense since it would be damn hard to create a hover I think )
if you look @ the function its is brought in in the function variables but then does nothing.
image
image
image

I'm open as to what to do because clearly I'd have TONS of icon,icon to change to icon, if I'm to do it 100% correct.
image

As it stands its totally broken.
in 1.5x you'll see its used as an override 
https://github.com/joomla/joomla-cms/issues/30211#issuecomment-665497073

In 3.x you'll see its being used as ??
image
image

THAT would be extremely easy to implement though I would still have tons of icon,icon to clean up.

**Comment by N6REJ** [comment]
@wilsonge thoughts?

**Comment by N6REJ** [comment]
After discussing @ length with @SharkyKZ we decided to leave the variable but deprecate its behavior.

**Comment by hans2103** [comment]
People might call for `JToolbarHelper::custom('something', 'something', '', 'alt-text');`. Removing it now will break b/c. 
To make it deprecate is a good point. You can safely remove it in the next J version.

**Comment by brianteeman** [comment]
This should be closed https://github.com/joomla/joomla-cms/pull/31211#issuecomment-715010060 and a new pr opened for the deprecation

**Comment by N6REJ** [comment]
> This should be closed [#31211 (comment)](https://github.com/joomla/joomla-cms/pull/31211#issuecomment-715010060) and a new pr opened for the deprecation

Thats what this is!

**Comment by brianteeman** [comment]
It is now you changed the title :)

File: administrator/components/com_cache/src/View/Cache/HtmlView.php [hunk 1]
```diff
@@ -125,8 +125,8 @@ protected function addToolbar(): void
 		// Get the toolbar object instance
 		$toolbar = Toolbar::getInstance('toolbar');
 
-		ToolbarHelper::custom('delete', 'delete.png', 'delete_f2.png', 'JTOOLBAR_DELETE', true);
-		ToolbarHelper::custom('deleteAll', 'remove.png', 'delete_f2.png', 'JTOOLBAR_DELETE_ALL', false);
+		ToolbarHelper::custom('delete', 'delete.png', '', 'JTOOLBAR_DELETE', true);
```
Comment (Quy) [inline administrator/components/com_cache/src/View/Cache/HtmlView.php]:
```suggestion
		ToolbarHelper::custom('delete', 'delete', '', 'JTOOLBAR_DELETE', true);
```
Remove extension. Repeat in other instances.

File: libraries/src/Toolbar/ToolbarHelper.php [hunk 2]
```diff
@@ -99,7 +99,7 @@ public static function divider()
 	 *
 	 * @since   1.5
 	 */
-	public static function custom($task = '', $icon = '', $iconOver = '', $alt = '', $listSelect = true, $formId = null)
+	public static function custom($task = '', $icon = '', string $iconOver = '', $alt = '', $listSelect = true, $formId = null)
```
Comment (Quy) [inline libraries/src/Toolbar/ToolbarHelper.php]:
```suggestion
	public static function custom($task = '', $icon = '', $iconOver = '', $alt = '', $listSelect = true, $formId = null)
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 29a54dba07d17486bb758eb441944107736022ed<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31211">issues.joomla.org/tracker/joomla-cms/31211</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 70cbc49e2b2777ade8a33b0b24d50fa6f4b9036c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31211">issues.joomla.org/tracker/joomla-cms/31211</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31211">issues.joomla.org/tracker/joomla-cms/31211</a>.</sub>

**Comment by richard67** [comment]
Thanks!

**Pull request merged**